Some compilers will fail to build with "undeclared identifier 'digit_to_number'" error

There are some compilers that fail to compile when building version 7.7.0.

This problem was encountered in a CI build failure of the package manager conan (https://github.com/conan-io/conan-center-index/pull/8761#issuecomment-1012788946).

The following is the error log when building with Clang.

<details>
<summary>CI Failure log</summary>

```
[...]
[ 14%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/connection.cxx.o
[ 18%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/cursor.cxx.o
[ 22%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/encodings.cxx.o
[ 25%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/errorhandler.cxx.o
[ 29%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/except.cxx.o
[ 33%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/field.cxx.o
[ 37%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/largeobject.cxx.o
[ 40%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/notification.cxx.o
[ 44%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/params.cxx.o
[ 48%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/pipeline.cxx.o
[ 51%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/result.cxx.o
[ 55%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/robusttransaction.cxx.o
[ 59%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/row.cxx.o
[ 62%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/sql_cursor.cxx.o
[ 66%] Building CXX object source_subfolder/src/CMakeFiles/pqxx.dir/strconv.cxx.o
source_subfolder/src/CMakeFiles/pqxx.dir/build.make:302: recipe for target 'source_subfolder/src/CMakeFiles/pqxx.dir/strconv.cxx.o' failed
CMakeFiles/Makefile2:151: recipe for target 'source_subfolder/src/CMakeFiles/pqxx.dir/all' failed
Makefile:170: recipe for target 'all' failed
libpqxx/7.7.0: 
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


/home/conan/w/BuildSingleReference/.conan/data/libpqxx/7.7.0/_/_/build/e52f527ce44f8ad45ccc34f94acd11eaa754d477/source_subfolder/src/strconv.cxx:441:46: error: use of undeclared identifier 'digit_to_number'; did you mean 'pqxx::internal::digit_to_number'?
      result = absorb_digit_positive(result, digit_to_number(data[i]));
                                             ^~~~~~~~~~~~~~~
                                             pqxx::internal::digit_to_number
/home/conan/w/BuildSingleReference/.conan/data/libpqxx/7.7.0/_/_/build/e52f527ce44f8ad45ccc34f94acd11eaa754d477/source_subfolder/include/pqxx/internal/conversions.hxx:34:15: note: 'pqxx::internal::digit_to_number' declared here
constexpr int digit_to_number(char c) noexcept
              ^
/home/conan/w/BuildSingleReference/.conan/data/libpqxx/7.7.0/_/_/build/e52f527ce44f8ad45ccc34f94acd11eaa754d477/source_subfolder/src/strconv.cxx:455:46: error: use of undeclared identifier 'digit_to_number'; did you mean 'pqxx::internal::digit_to_number'?
      result = absorb_digit_negative(result, digit_to_number(data[i]));
                                             ^~~~~~~~~~~~~~~
                                             pqxx::internal::digit_to_number
/home/conan/w/BuildSingleReference/.conan/data/libpqxx/7.7.0/_/_/build/e52f527ce44f8ad45ccc34f94acd11eaa754d477/source_subfolder/include/pqxx/internal/conversions.hxx:34:15: note: 'pqxx::internal::digit_to_number' declared here
constexpr int digit_to_number(char c) noexcept
              ^
2 errors generated.
make[2]: *** [source_subfolder/src/CMakeFiles/pqxx.dir/strconv.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [source_subfolder/src/CMakeFiles/pqxx.dir/all] Error 2
make: *** [all] Error 2
libpqxx/7.7.0: WARN: Build folder is dirty, removing it: /home/conan/w/BuildSingleReference/.conan/data/libpqxx/7.7.0/_/_/build/e52f527ce44f8ad45ccc34f94acd11eaa754d477
libpqxx/7.7.0: ERROR: Package 'e52f527ce44f8ad45ccc34f94acd11eaa754d477' build failed
libpqxx/7.7.0: WARN: Build folder /home/conan/w/BuildSingleReference/.conan/data/libpqxx/7.7.0/_/_/build/e52f527ce44f8ad45ccc34f94acd11eaa754d477
ERROR: libpqxx/7.7.0: Error in build() method, line 101
	cmake.build()
	ConanException: Error 2 while ex
```
</details>

--- a/src/strconv.cxx
+++ b/src/strconv.cxx
@@ -438,7 +438,8 @@ template<typename T>
   if (pqxx::internal::is_digit(initial))
   {
     for (; pqxx::internal::is_digit(data[i]); ++i)
-      result = absorb_digit_positive(result, digit_to_number(data[i]));
+      result = absorb_digit_positive(
+        result, pqxx::internal::digit_to_number(data[i]));
   }
   else if (initial == '-')
   {
@@ -452,7 +453,8 @@ template<typename T>
         "Converting string to " + pqxx::type_name<T> +
         ", but it contains only a sign."};
     for (; i < std::size(text) and pqxx::internal::is_digit(data[i]); ++i)
-      result = absorb_digit_negative(result, digit_to_number(data[i]));
+      result = absorb_digit_negative(
+        result, pqxx::internal::digit_to_number(data[i]));
   }
   else
   {

